src/reactor.cc

```diff
@@ -0,0 +1,406 @@
+  DecayInventory(tritium_reserve);
+  DecayInventory(tritium_storage);
+
+  ExtractHelium(tritium_core);
+  ExtractHelium(tritium_reserve);
+  ExtractHelium(tritium_storage);
+
+  // Pull excesss tritium out of reserve and put it into storage
+  // Replenish core from Helium Extraction.
+  if (!tritium_reserve.empty() && core_loaded) {
+    double core_deficit = startup_inventory - tritium_core.quantity();
+
+    // Excess tritium will always be in tritium_reserve.
+    double surplus = std::max(
+        tritium_reserve.quantity() - reserve_inventory - core_deficit, 0.0);
+
+    cyclus::Material::Ptr reserve_fuel = tritium_reserve.Pop();
+    cyclus::Material::Ptr core_fuel = tritium_core.Pop();
+
+    core_fuel->Absorb(reserve_fuel->ExtractQty(core_deficit));
+    tritium_storage.Push(reserve_fuel->ExtractQty(surplus));
+
+    RecordOperationalInfo(
+        "Tritium Moved",
+        std::to_string(core_deficit) + "kg of T moved from reserve to core");
+    if (surplus > 0.0) {
+      RecordOperationalInfo(
+          "Tritium Moved",
+          std::to_string(surplus) + "kg of T moved from reserve to storage");
+    }
+    tritium_reserve.Push(reserve_fuel);
+    tritium_core.Push(core_fuel);
+
+    CombineInventory(tritium_storage);
+  }
+
+  // This pulls out some of the blanket each timestep so that fresh blanket can
+  // be added.
+  if (!blanket.empty() &&
+      blanket.quantity() >= blanket_size * blanket_turnover_rate) {
+    cyclus::Material::Ptr blanket_mat = blanket.Pop();
+    cyclus::Material::Ptr spent_blanket =
+        blanket_mat->ExtractQty(blanket_size * blanket_turnover_rate);
+    RecordOperationalInfo(
+        "Blanket Cycled",
+        std::to_string(spent_blanket->quantity()) + "kg of blanket removed");
+    blanket.Push(blanket_mat);
+  } else if (!blanket.empty() &&
+             blanket.quantity() < blanket_size * blanket_turnover_rate) {
+    RecordOperationalInfo(
+        "Blanket Not Cycled",
+        "Total blanket material (" + std::to_string(blanket.quantity()) +
+            ") insufficient to extract " +
+            std::to_string(blanket_size * blanket_turnover_rate) + "kg!");
+  }
+}
+
+void Reactor::Tock() {
+  if (!core_loaded) {
+    try {
+      Startup();
+      fuel_startup_policy.Stop();
+      // blanket_startup_policy.Stop();
+      fuel_refill_policy.Start();
+      core_loaded = true;
+    } catch (const std::exception& e) {
+      RecordOperationalInfo("Startup Error", e.what());
+      LOG(cyclus::LEV_INFO2, "Reactor") << e.what();
+    }
+  }
+
+  CombineInventory(tritium_reserve);
+  CombineInventory(blanket);
+
+  RecordInventories(tritium_core.quantity(), tritium_reserve.quantity(),
+                    tritium_storage.quantity(), blanket.quantity(),
+                    helium_storage.quantity());
+}
+
+void Reactor::EnterNotify() {
+  cyclus::Facility::EnterNotify();
+
+  fuel_startup_policy
+      .Init(this, &tritium_reserve, std::string("Tritium Storage"),
```

Comment:
Yeah, this results in behavior identical to initializing without an inventory policy. Material is always ordered whenever less than reserve+startup is on-hand, up to reserve+startup. This is equivalent to a max inventory size of reserve+startup without any throughput restrictions (the space remaining is always the request size)

Reply:
Is this something that needs to be changed, or just a comment that it was a funky way for me to have implemented it? I ask because this was the way that seemed the most obvious to me, and my understanding is that it works the way I want it to (as long as inventory < startup+reserve, try to buy however much fuel will get inventory = startup+reserve). It's worth noting that I've changed the specific requirements on this a little bit, but the overall way it works is still the same. Happy to change it if it needs to go, but I'm not totally sure what I would change it to if I need to, so just checking.